### ISSUE TITLE
Safe\DateTimeImmutable is not compatible with \DatePeriod

### ISSUE THREAD

**Issue by bcremer** [thecodingmachine/safe#181]
The `Safe\`  Versions of `DateTime(Immutable)` cannot be used with PHPs inbuilt `\DatePeriod` class:

## PHPs inbuilt DateTime(immutable)
```php
$datePeriod = new \DatePeriod(
    new \DateTimeImmutable('2019-01-01'),
    \DateInterval::createFromDateString('1 day'),
    new \DateTimeImmutable('2019-01-04')
);
var_dump(iterator_count($datePeriod)); // int(3)
```

## Safe\DateTime(immutable)
```php
$datePeriod = new \DatePeriod(
    new \DateTimeImmutable('2019-01-01'),
    \DateInterval::createFromDateString('1 day'),
    new \DateTimeImmutable('2019-01-04')
);
var_dump(iterator_count($datePeriod)); // int(0)
```

My guest is that `\DatePeriod` uses direct access to the internal properties of the `DateTime(Immutable)` and thus works around `innerDateTime`.

A possible fix could be to synchronize the objects properties with the `innerDateTime` property.

**Comment by bcremer** [comment]
This has already been fixed on `master` with https://github.com/thecodingmachine/safe/pull/180.
Closing here.

**Comment by jdreesen** [comment]
Did you try it on master? There was a DateTime fix yesterday: #179, #180

**Comment by bcremer** [comment]
Yes, I was just about to write nearly the same fix when I saw it was already fixed in the master branch :)
